jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletContextHandler.java

```diff
@@ -696,6 +696,27 @@ else if (handler instanceof ServletHandler)
         relinkHandlers();
     }
 
+    /**
+     * Utility Method to allow for manual execution of {@link javax.servlet.ServletContainerInitializer} when using Embedded Jetty.
+     * @param containerInitializer the ServletContainerInitializer to register.
+     * @see Initializer
+     */
+    public void addServletContainerInitializer(ServletContainerInitializer containerInitializer)
+    {
+        addServletContainerInitializer(containerInitializer, Collections.emptySet());
+    }
+
+    /**
+     * Utility Method to allow for manual execution of {@link javax.servlet.ServletContainerInitializer} when using Embedded Jetty.
+     * @param containerInitializer the ServletContainerInitializer to register.
+     * @param classes the Set of application classes.
+     * @see Initializer
+     */
+    public void addServletContainerInitializer(ServletContainerInitializer containerInitializer, Set<Class<?>> classes)
+    {
+        addManaged(new Initializer(this, containerInitializer, classes));
```

Comment:
You should not be able to add a ServletContainerInitializer after the ServletContextHandler has started.

But in any case I'm -1 to add these methods to ServletContextHandler without a greater consideration of the integration with the handling of ServletContainerInitializers by the AnnotationConfiguration.  There's already been divergence with the addition of the ServletContextHandler.Initializer and the ServletContextHandler.ServletContainerInitializerCaller.  Once we expose methods, it's extremely hard to remove them, so we have to be sure we're crafting exactly the right methods.

Reply:
Personally, I *love* these new API methods.
It makes things far easier for embedded users.
These methods  are also use to expand upon for issue #5834 (after this issue is merged)

Reply:
Sure, I agree it looks attractive. However, it does not take into account anything like ordering of SCIs that is done by the AnnotationConfiguration that winds up using a _completely different_ mechanism to start the SCIs:  I'd like us to have a single mechanism to order and start the SCIs that can be leveraged by both the plain ServletContextHandler _and_ the AnnotationConfiguration.

Reply:
@janbartel I agree that it would be ideal to have just one mechanism, but I also think that this is probably out of scope for this PR.

The current mechanism for this is baked into `AnnotationConfiguration` and that is not required when just using embedded Jetty. So `AnnotationConfiguration` and other usages of this would need to use the mechanism in jetty-servlet.

Reply:
It's not out of scope for this PR: we've already had 2 or 3 bites at the SCI cherry without unifying the approach, we should not have another.

Reply:
+1.  I know it feels like a bit of a pile on.... but let's get this mechanism right this time.    New API is good, but a single mechanism behind should be possible and will avoid future problems like this.